drv/auxflash-server/src/main.rs

```diff
@@ -434,16 +434,36 @@ impl NotificationHandler for ServerImpl {
 
 fn scan_for_active_slot(qspi: &Qspi) -> Option<u32> {
     for i in 0..SLOT_COUNT {
-        if let Ok(chck) = read_slot_checksum(qspi, i) {
-            if chck.0 == AUXI_CHECKSUM {
-                return Some(i);
-            }
+        let handle = SlotReader {
+            qspi,
+            base: i * SLOT_SIZE as u32,
+        };
+
+        let Ok(chck) = handle.read_stored_checksum() else {
+            // Just skip to the next slot if it's empty or invalid.
+            continue;
+        };
+
+        if chck.0 != AUXI_CHECKSUM {
+            // If it's not the chunk we're interested in, don't bother hashing
+            // it.
+            continue;
+        }
+
+        let Ok(actual) = handle.calculate_checksum() else {
+            // TODO: this ignores I/O errors, but, this is how the code has
+            // always been structured...
+            continue;
+        };
+
+        if chck == actual {
```

Comment:
Not sure if there are already ringbuf logs in this task, but - should we log if this condition is false? It seems like that would be very surprising.

Reply:
I'm not sure that it's that surprising; a difference between actual and claimed checksum could occur if a write is aborted partway through, for example.